org-cyf-sdc/content/tools/sprints/3/process_spelunking_workshop.md

```diff
@@ -0,0 +1,539 @@
+# Workshop: Process Spelunking
+
+This workshop will teach you about how the programs you write, interact with files.
+And in the process, we'll learn more about the kernel, and how it helps your programs do things.
+
+## Preparation
+
+### Linux
+
+In order to participate in this workshop, you will need a linux computer.
+I recommend using an ubuntu VM in AWS (free tier instance), or alternatively a VM inside virtual box.
+
+> [!WARNING]
+>
+> Docker linux will not work properly.
+> Don't use it.
+> (If you don't know what docker is, you're probably grand.)
+
+<!--
+  Should we warn against WoL-subsystem (linux emulator on windows) too?
+  -->
+
+For people who run linux on their own laptop, I still recommend an AWS instance,
+so you won't get distracted by all the other programs running under your user account.
+
+### Preparing Your Linux
+
+On the linux computer, make sure the following packages are installed:
+- `python3`
+- `lsof`
+- `strace`
+
+> [!TIP]
+>
+> On ubuntu, you can run `sudo apt install python3 lsof strace` to get these programs installed.
+
+<!--
+  Beyond this point, `sudo` isn't required, but I see no reason to deny it.
+  Everything (except for the package installation) can be done as a normal user.
+  -->
+
+
+
+### Linux Skills
+
+I recommend you learn how to use the `less` program.
+
+Confirm that if you run `man ps`, you get shown a man-page.
+(It's a lot of text, you can exit it with `q`.)
+
+Make sure that you can:
+- create a new text file on the linux machine,
```

Comment:
```suggestion
- create a new text file on the linux machine (e.g. by running the `touch` command),
```

Reply:
While I agree that the `touch` command can be useful, if creating empty files is the extend of them being able to create files, they won't succeed the exercise.
Since in multiple occasions they need to create small files with code.

My idea was that they know how to use at least one editor.
If they're taught an editor (vim, nano, emacs, something else?), that'll be helpful.

Reply:
Makes sense - feel free to dismiss this suggestion - people at least should know how to use VS Code, and hopefully also `echo`. No promises about vim/nano/emacs though!